AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoFrameResender.swift

```diff
@@ -13,88 +13,98 @@ import CoreMedia
-    private var resendTimer: DispatchSourceTimer?
+    private let resendTimer: DispatchSourceTimer
+    
     // Cached constant values
-    private let resendTimeInterval: CMTime
     private let resendScheduleLeewayMs = DispatchTimeInterval.milliseconds(20)
-
+    
+    private let logger: Logger
+    
+    private let lock = NSLock()
+    
     private var lastSendTimestamp: CMTime?
     private var lastVideoFrame: VideoFrame?
 
-    private let resendFrameHandler: (VideoFrame) -> Void
-
     /// Callback will be triggered with a previous `VideoFrame` which will have different timestamp
     /// than originally sent with so it won't be dropped by downstream encoders
-    init(minFrameRate: UInt, resendFrameHandler: @escaping (VideoFrame) -> Void) {
-        self.minFrameRate = minFrameRate
-        self.resendFrameHandler = resendFrameHandler
-        self.resendTimeInterval = CMTime(value: CMTimeValue(Constants.millisecondsPerSecond / Int(minFrameRate)),
-                                      timescale: CMTimeScale(Constants.millisecondsPerSecond))
-    }
-
-    func stop() {
-        resendTimer?.cancel()
-        resendTimer = nil
-    }
-
-    /// Calling this function will kick off a timer which will begin checking if frames need to be resent
-    /// to maintain a minimum frame frame
-    func frameDidSend(videoFrame: VideoFrame) {
-        lastSendTimestamp = CMClockGetTime(CMClockGetHostTimeClock())
-        lastVideoFrame = videoFrame
-
-        if let resendTimer = resendTimer,
-           resendTimer.isCancelled == false {
-            // There is already a timer running
-            return
-        }
-        let timer = DispatchSource.makeTimerSource(flags: .strict, queue: resendQueue)
-        resendTimer = timer
-
-        // This timer is invoked every resendTimeInterval when no frame is sent from video source
-        timer.setEventHandler(handler: { [weak self] in
+    init(minFrameRate: UInt,
+         logger: Logger,
+         resendFrameHandler: @escaping (VideoFrame) -> Void) {
+        self.resendTimer = DispatchSource.makeTimerSource(flags: .strict, queue: resendQueue)
+        self.logger = logger
+        
+        super.init()
+        
+        let minFrameInterval = Constants.millisecondsPerSecond / Int(minFrameRate)
+        let resendTimeInterval:CMTime = CMTime(value: CMTimeValue(minFrameInterval),
+                                         timescale: CMTimeScale(Constants.millisecondsPerSecond))
+        
+        self.resendTimer.setEventHandler(handler: { [weak self] in
             guard let `self` = self else {
-                timer.cancel()
+                self?.resendTimer.cancel()
+                self?.logger.error(msg: "Unable to resend video frame, VideoFrameResender is unavailable. Thread: \(Thread.current)")
                 return
             }
-
+            
+            self.logger.info(msg: "Checking if there is pending frame for resending. Thread: \(Thread.current)")
```

Comment:
These logs are pretty verbose for `info` level. Probably should be `debug`? Also could we just add the Thread logging to the logger class? Seems clunky to do it adhoc.

Reply:
Will do!